sky/cloud_stores.py

```diff
@@ -219,8 +219,77 @@ def get_storage_from_path(url: str) -> CloudStorage:
     return _REGISTRY[result.scheme]
 
 
+class IBMCosCloudStorage(CloudStorage):
+    """IBM Cloud Storage."""
+    # install rclone if package isn't already installed
+    # pylint: disable=line-too-long
+    _GET_RCLONE = [
+        'rclone version >/dev/null 2>&1 || curl https://rclone.org/install.sh | sudo bash',
+    ]
+
+    def is_directory(self, url: str) -> bool:
+        """Returns whether IBM cos bucket's 'url' is a directory.
+
+        In cloud object stores, a "directory" refers to a regular object whose
+        name is a prefix of other objects.
+        """
+
+        region = url.split('//')[1].split('/')[0]
+        s3 = ibm.get_cos_resource(region)
+        bucket_name, path, _ = data_utils.split_cos_path(url)
+        bucket = s3.Bucket(bucket_name)
+
+        num_objects = 0
+        for obj in bucket.objects.filter(Prefix=path):
+            print(obj)
+
+            num_objects += 1
+            if obj.key == path:
+                return False
+            # If there are more than 1 object in filter, then it is a directory
+            if num_objects == 3:
+                return True
+
+        # A directory with few or no items
+        return True
+
+    def _get_rclone_sync_command(self, source: str, destination: str):
+        bucket_name, data_path, bucket_region = data_utils.split_cos_path(
+            source)
+        bucket_rclone_profile = data_utils.Rclone.get_rclone_bucket_profile(
+            bucket_name, 'IBM')
```

Comment:
I agree using random strings is definitely not the way, and gave it thought as well.
The problem i see with this approach is that to verify the values on `data_utils.py` i'll need to import `storage.py`. It will create a circular import. I suggest creating an enum in `data_utils.Rclone` containing the rclone supported storage providers. `cloud` arg on Rclone's methods will be of that type.